runtime-transaction/benches/process_compute_budget_instructions.rs

```diff
@@ -0,0 +1,141 @@
+#![feature(test)]
+extern crate test;
```

Comment:
would be nice, in my opinion, to use criterion so we don't need to use nightly AND we can get the throughput numbers in terms of TPS.

See https://github.com/anza-xyz/agave/pull/2262 as a recent example of using criterion for benchmark w/ throughput.

Reply:
nice to do it without nightly, 6f922c6cbce059c8518a88e967769c8052b1f987

Reply:
<details>
<summary>bench result with Throughput</summary>

```
bench_size_1024/No instructions
                        time:   [3.5170 µs 3.5229 µs 3.5279 µs]
                        thrpt:  [290.26 Melem/s 290.67 Melem/s 291.16 Melem/s]

bench_size_1024/No builtins
                        time:   [16.493 µs 16.504 µs 16.516 µs]
                        thrpt:  [62.002 Melem/s 62.046 Melem/s 62.087 Melem/s]
Found 4 outliers among 100 measurements (4.00%)
  1 (1.00%) low mild
  1 (1.00%) high mild
  2 (2.00%) high severe

bench_size_1024/Only compute-budget instructions
                        time:   [27.893 µs 27.934 µs 27.978 µs]
                        thrpt:  [36.601 Melem/s 36.658 Melem/s 36.712 Melem/s]
Found 15 outliers among 100 measurements (15.00%)
  4 (4.00%) low mild
  8 (8.00%) high mild
  3 (3.00%) high severe

bench_size_1024/Only builtins
                        time:   [16.859 µs 16.868 µs 16.877 µs]
                        thrpt:  [60.676 Melem/s 60.708 Melem/s 60.739 Melem/s]
Found 17 outliers among 100 measurements (17.00%)
  1 (1.00%) low severe
  2 (2.00%) low mild
  7 (7.00%) high mild
  7 (7.00%) high severe

bench_size_1024/Mixed instructions
                        time:   [447.94 µs 449.48 µs 451.58 µs]
                        thrpt:  [2.2676 Melem/s 2.2782 Melem/s 2.2860 Melem/s]
```

</details>

It'd be better if Criterion supports customizing unit label from `Melem/s` to `M tps` 🫠

Reply:
I think the `bench_size_1024` is not really that instructive, the number of txs per iteration doesn't mattter afaict (and we really re-use the same one anyway?).

Would probably be more descriptive to have the number of instructions, as that is highly relevant to throughput, right?

Reply:
Feels a bit redundant to have to name the `fn` itself, then benchmark_group, then bench_function. I settled with setting benchmark_group's name to be same as `fn` name (as each `fn bench...()` is physically a group), then add instruction count as bench_function name. The report reads better.

<details>

```
bench_process_compute_budget_instructions_empty/0 instructions
                        time:   [3.5387 µs 3.5400 µs 3.5415 µs]
                        thrpt:  [289.15 Melem/s 289.27 Melem/s 289.37 Melem/s]
Found 15 outliers among 100 measurements (15.00%)
  7 (7.00%) high mild
  8 (8.00%) high severe

bench_process_compute_budget_instructions_no_builtins/4 dummy Instructions
                        time:   [14.143 µs 14.150 µs 14.157 µs]
                        thrpt:  [72.331 Melem/s 72.369 Melem/s 72.401 Melem/s]
Found 8 outliers among 100 measurements (8.00%)
  1 (1.00%) high mild
  7 (7.00%) high severe

bench_process_compute_budget_instructions_compute_budgets/4 compute-budget instructions
                        time:   [24.473 µs 24.501 µs 24.535 µs]
                        thrpt:  [41.737 Melem/s 41.795 Melem/s 41.842 Melem/s]
Found 5 outliers among 100 measurements (5.00%)
  1 (1.00%) low mild
  3 (3.00%) high mild
  1 (1.00%) high severe

bench_process_compute_budget_instructions_builtins/4 dummy builtins
                        time:   [14.144 µs 14.150 µs 14.157 µs]
                        thrpt:  [72.333 Melem/s 72.367 Melem/s 72.397 Melem/s]
Found 6 outliers among 100 measurements (6.00%)
  1 (1.00%) high mild
  5 (5.00%) high severe

bench_process_compute_budget_instructions_mixed/1024 mixed instructions
                        time:   [2.5592 ms 2.5645 ms 2.5714 ms]
                        thrpt:  [398.23 Kelem/s 399.30 Kelem/s 400.13 Kelem/s]
Found 10 outliers among 100 measurements (10.00%)
  4 (4.00%) high mild
  6 (6.00%) high severe
```

</details>